backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -0,0 +1,215 @@
+                            switch (endpoint.getType()) {
+                                case CAMEL:
+                                case WEBHOOK:
+                                case EMAIL_SUBSCRIPTION:
+                                    return session.persist(endpoint.getProperties())
+                                            .onItem().call(session::flush);
+                                default:
+                                    // Do nothing.
+                                    break;
+                            }
+                        }
+                        /*
+                         * If this line is reached, it means the endpoint properties are null or we don't support
+                         * persisting properties for the endpoint type. We still have to return something.
+                         */
+                        return Uni.createFrom().voidItem();
+                    })
+                    .replaceWith(endpoint);
+        });
+    }
+
+    public Uni<List<Endpoint>> getEndpointsPerType(String accountId, Set<EndpointType> type, Boolean activeOnly, Query limiter, boolean useStatelessSession) {
+        return commonStateSessionFactory.withSession(useStatelessSession, session -> {
+            // TODO Modify the parameter to take a vararg of Functions that modify the query
+            // TODO Modify to take account selective joins (JOIN (..) UNION (..)) based on the type, same for getEndpoints
+            String query = "SELECT e FROM Endpoint e WHERE e.type IN (:endpointType)";
+            if (accountId == null) {
+                query += " AND e.accountId IS NULL";
+            } else {
+                query += " AND e.accountId = :accountId";
+            }
+            if (activeOnly != null) {
+                query += " AND enabled = :enabled";
+            }
+
+            if (limiter != null) {
+                query = limiter.getModifiedQuery(query);
+            }
+
+            Mutiny.Query<Endpoint> mutinyQuery = session.createQuery(query, Endpoint.class)
+                    .setParameter("endpointType", type);
+
+            if (accountId != null) {
+                mutinyQuery = mutinyQuery.setParameter("accountId", accountId);
+            }
+            if (activeOnly != null) {
+                mutinyQuery = mutinyQuery.setParameter("enabled", activeOnly);
+            }
+
+            if (limiter != null && limiter.getLimit() != null && limiter.getLimit().getLimit() > 0) {
+                mutinyQuery = mutinyQuery.setMaxResults(limiter.getLimit().getLimit())
+                        .setFirstResult(limiter.getLimit().getOffset());
+            }
+
+            return mutinyQuery.getResultList()
+                    .onItem().call(endpoints -> loadProperties(endpoints, useStatelessSession));
+        });
+    }
+
+    public Uni<Endpoint> getOrCreateEmailSubscriptionEndpoint(String accountId, EmailSubscriptionProperties properties, boolean useStatelessSession) {
```

Comment:
This now exists in both `EndpointResources` and `EndpointRepository` (same code). `EndpointRepository` will be refactored later to make the code simpler.